plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/validator/URLValidator.java

```diff
@@ -0,0 +1,53 @@
+/******************************************************************************* 
+ * Copyright (c) 2016 Red Hat, Inc. 
+ * Distributed under license by Red Hat, Inc. All rights reserved. 
+ * This program is made available under the terms of the 
+ * Eclipse Public License v1.0 which accompanies this distribution, 
+ * and is available at http://www.eclipse.org/legal/epl-v10.html 
+ * 
+ * Contributors: 
+ * Red Hat, Inc. - initial API and implementation 
+ ******************************************************************************/
+package org.jboss.tools.openshift.internal.ui.validator;
+
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.core.databinding.validation.IValidator;
+import org.eclipse.core.databinding.validation.ValidationStatus;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.osgi.util.NLS;
+import org.jboss.tools.openshift.common.core.utils.UrlUtils;
+
+/**
+ * Validator for validating URLs in the UI
+ * @author jeff.cantrill
+ *
+ */
+public class URLValidator implements IValidator{
+	
+	private final String invalidURLMessage;
+	private boolean allowEmpty = false;
+	
+	/**
+	 * @param urlType  The value to plug into error message of 'Please provide a valid TYPE URL'
+	 * @param allowEmpty  Empty/Blank values will return OK
+	 */
+	public URLValidator(String urlType, boolean allowEmpty) {
+		invalidURLMessage = NLS.bind("Please provide a valid {0} URL", urlType);
+		this.allowEmpty  = allowEmpty;
+	}
+	
+	
+
+	@Override
+	public IStatus validate(Object in) {
+		String value = (String) in;
```

Comment:
There is a org.apache.commons.validator.routines.UrlValidator provided by commons-validator